src/google/protobuf/compiler/cpp/cpp_helpers.h

```diff
@@ -74,6 +74,11 @@ string SuperClassName(const Descriptor* descriptor);
 // anyway, so normally this just returns field->name().
 string FieldName(const FieldDescriptor* field);
 
+// Get the sanitized name that should be used for the  given enum in C++ code.
+// The name is optionally mangled for reserved keyword collisions. (Nested
+// names need not be mangled as they receive a prefix from the outer message type.)
+string EnumName(const EnumValueDescriptor* enum_value, bool mangle);
```

Comment:
Are you sure you want it to always be mangled?

That means we go from e.g. `TestConflictingEnumNames_NestedConflictingEnum_and` to `TestConflictingEnumNames_NestedConflictingEnum_and_` ?